app/controllers/individual_market/exchanges/hbx_profiles_controller.rb

```diff
@@ -0,0 +1,162 @@
+        message_params = {
+          sender_id: hbx_profile.id,
+          parent_message_id: hbx_profile.id,
+          from: 'Plan Shopping Web Portal',
+          to: 'Agent Mailbox',
+          subject: "Please contact #{details[:first_name]} #{details[:last_name]}.",
+          body: body
+        }
+
+        # Send secure messages
+        send_secure_messages(message_params, hbx_profile, agent)
+
+        # Send email notification
+        send_email_notification(
+          agent: agent,
+          role: role,
+          first_name: details[:first_name],
+          full_name: details[:full_name],
+          email: details[:email],
+          person_present: params[:person].present?
+        )
+      end
+
+      def extract_and_sanitize_user_details(params)
+        if params[:person].present?
+          insured = Person.find(params[:person])
+          {
+            first_name: sanitize_html(insured.first_name),
+            last_name: sanitize_html(insured.last_name),
+            full_name: sanitize_html(insured.full_name),
+            email: sanitize_html(insured.emails.last.try(:address) || insured.try(:user).try(:email))
+          }
+        else
+          {
+            first_name: sanitize_html(params[:first_name]),
+            last_name: sanitize_html(params[:last_name]),
+            full_name: sanitize_html("#{params[:first_name]} #{params[:last_name]}"),
+            email: sanitize_html(params[:email])
+          }
+        end
+      end
+
+      def construct_message_body(details, params)
+        if params[:person].present?
+          root = sanitize_html("http://#{request.env['HTTP_HOST']}/exchanges/agents/resume_enrollment?person_id=#{params[:person]}&original_application_type:")
+          <<~HTML
+            Please contact #{details[:first_name]} #{details[:last_name]}. <br>
+            Plan shopping help has been requested by #{details[:email]} <br>
+            <a href="#{root}phone">Assist Customer</a> <br>
+          HTML
+        else
+          <<~HTML
+            Please contact #{details[:first_name]} #{details[:last_name]}. <br>
+            Plan shopping help has been requested by #{details[:email]} <br>
+          HTML
+        end
+      end
+
+      def send_secure_messages(message_params, hbx_profile, agent)
+        create_secure_message(message_params, hbx_profile, :sent)
+        create_secure_message(message_params, agent, :inbox)
+      end
+
+      def send_email_notification(params)
+        result = UserMailer.new_client_notification(
+          find_email(params[:agent], params[:role]),
+          params[:first_name],
+          params[:full_name],
+          params[:role],
+          params[:email],
+          params[:person_present]
+        )
+        result.deliver_now
+      end
+
+      def call_customer_service(first_name, last_name)
+        "No match found for #{first_name} #{last_name}.  Please call Customer Service at: (855)532-5465 for assistance.<br/>"
```

Comment:
this should be a separate partial probably or something on the front end, I'm approving b/c of business urgency